components/ModFacilitySearchbar.vue

```diff
@@ -0,0 +1,181 @@
+<template>
+    <div
+        class="flex flex-col gap-1 w-fit"
+    >
+        <div
+            class="flex items-center px-3 py-3.5 w-96 h-12 bg-secondary-bg
+            rounded-lg border border-primary-text-muted text-primary-text
+            text-sm font-normal font-sans placeholder-primary-text-muted
+            outline outline-0 -outline-offset-2 gap-2 divide-x-2
+            focus-within:outline-currentColor
+            focus-within:outline-2
+            "
+        >
+            <!-- TODO: Change i18n key for the placeholder if necessary -->
+            <input
+                ref="searchInputElement"
+                v-model="searchInput"
+                type="text"
+                :placeholder="$t('modFacilitySearchbar.placeholderTextFacilitySearchbar')"
+                class="grow focus-visible:outline-none"
+                @blur="handleSearchInputBlur"
+                @keydown.esc="handleSearchInputEsc"
+                @keydown.down="handleSearchInputArrowDown"
+                @keydown.up="handleSearchInputArrowUp"
+                @keydown.enter="handleSearchInputEnter"
+            >
+            <button
+                type="button"
+                @click="searchInputElement?.focus()"
+            >
+                <SVGLookingGlass
+                    role="img"
+                    title="searching icon"
+                    class="h-6 pl-2"
+                />
+            </button>
+        </div>
+        <div class="relative">
+            <ul
+                v-if="searchInput.trim() !== ''"
+                class="bg-white shadow-md border border-primary-gray-by rounded-lg absolute
+                w-full flex flex-col divide-y-2 overflow-hidden
+                "
+                :class="computedDropdownClass"
+            >
+                <!-- Search limited to 10 items -->
+                <li
+                    v-for="(facility, index) in filteredFacilities.slice(0, 10)"
+                    :key="facility.id"
+                    class="flex justify-between divide-x cursor-pointer"
+                    :class="[
+                        selectedFacilities.has(facility) && 'bg-primary-text-muted',
+                        selectedFacilityIndex === index ? 'bg-primary-hover text-primary-inverted' : 'opacity-95',
+                    ]"
+                    @click="(event) => { event.preventDefault(); handleListItemClick(facility) }"
+                    @mousedown="(event) => {
+                        // Prevent search input from being blurred
+                        event.preventDefault()
+                    }"
+                    @mousemove="selectedFacilityIndex = index"
```

Comment:
mousemove is a heavy event handler as it's triggered for every mouse pixel movement. Can we just use hover as a trigger here to set the id?